encoding/legacy-mb-japanese/iso-2022-jp/iso2022jp-encode-href-ascii.html

```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="iso-2022-jp"> <!-- test breaks if the server overrides this -->
+<title>ISO 2022-JP encoding (href)</title>
+<meta name="timeout" content="long">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="jis0208_index.js"></script>
+<script src="iso2022jp-encoder.js"></script>
+<link rel="author" title="Richard Ishida" href="mailto:ishida@w3.org">
+<link rel="help" href="https://encoding.spec.whatwg.org/#iso-2022-jp">
+<meta name="assert" content="The browser produces the expected byte sequences for all characters in the iso-2022-jp encoding after 0x9F when writing characters to an href value, using the encoder steps in the specification.">
+<script>
+function XnormalizeStr(str) {
+    str = encodeURIComponent(str);
+    var ptr = 0;
+    var out = "";
+    for (c = 0; c < str.length; c++) {
+        if (str.charAt(ptr) == "%") {
+            out += String.fromCodePoint(
+                parseInt(str.charAt(ptr + 1) + str.charAt(ptr + 2), 16)
+            );
+            ptr += 2;
+        } else out += str.charAt(ptr);
+        ptr++;
+    }
+}
+
+function encode(input, expected, desc) {
```

Comment:
Used in 5 test files; would be nice to factor it out but we can do that later.

Reply:
Yeah I think there's a general project to be done where you diff files in these encoding tests against each other and factor out large chunks, or use the .sub.html infrastructure to just make them into one test and then iframe N copies of them.